libgnucash/engine/mocks/gmock-Transaction.cpp

```diff
@@ -29,121 +29,105 @@ void
+    gnc_mocktransaction(trans)->begin_edit();
 }
 
 void
 xaccTransCommitEdit (Transaction *trans)
 {
     ASSERT_TRUE(GNC_IS_MOCKTRANSACTION(trans));
-    ((MockTransaction*)trans)->commit_edit();
+    gnc_mocktransaction(trans)->commit_edit();
 }
 
 Split *
 xaccTransGetSplit (const Transaction *trans, int i)
 {
-    EXPECT_TRUE(GNC_IS_MOCKTRANSACTION(trans));
-    if (GNC_IS_MOCKTRANSACTION(trans))
-        return ((MockTransaction*)trans)->get_split(i);
-    else
-        return nullptr;
+    SCOPED_TRACE("");
+    const MockTransaction* mocktrans = gnc_mocktransaction(trans);
+    return mocktrans ? mocktrans->get_split(i) : nullptr;
 }
 
 Split *
 xaccTransFindSplitByAccount(const Transaction *trans, const Account *acc)
 {
-    EXPECT_TRUE(GNC_IS_MOCKTRANSACTION(trans));
+    SCOPED_TRACE("");
+    const MockTransaction* mocktrans = gnc_mocktransaction(trans);
     EXPECT_TRUE(GNC_IS_MOCKACCOUNT(acc));
-    if (GNC_IS_MOCKTRANSACTION(trans))
-        return ((MockTransaction*)trans)->find_split_by_account(acc);
-    else
-        return nullptr;
+    return mocktrans ? mocktrans->find_split_by_account(acc) : nullptr;
 }
 
 time64
 xaccTransGetDate (const Transaction *trans)
 {
-    EXPECT_TRUE(GNC_IS_MOCKTRANSACTION(trans));
-    if (GNC_IS_MOCKTRANSACTION(trans))
-        return ((MockTransaction*)trans)->get_date();
-    else
-        return 0;
+    SCOPED_TRACE("");
+    const MockTransaction* mocktrans = gnc_mocktransaction(trans);
+    return mocktrans ? mocktrans->get_date() : 0;
 }
 
 void
 xaccTransSetDatePostedSecsNormalized (Transaction *trans, time64 time)
 {
     ASSERT_TRUE(GNC_IS_MOCKTRANSACTION(trans));
-    ((MockTransaction*)trans)->set_date_posted_secs_normalized(time);
+    gnc_mocktransaction(trans)->set_date_posted_secs_normalized(time);
 }
 
 const char *
 xaccTransGetDescription (const Transaction *trans)
 {
-    EXPECT_TRUE(GNC_IS_MOCKTRANSACTION(trans));
-    if (GNC_IS_MOCKTRANSACTION(trans))
-        return ((MockTransaction*)trans)->get_description();
-    else
-        return nullptr;
+    SCOPED_TRACE("");
+    const MockTransaction* mocktrans = gnc_mocktransaction(trans);
+    return mocktrans ? mocktrans->get_description() : "";
 }
 
 void
 xaccTransSetDescription (Transaction *trans, const char *desc)
 {
     ASSERT_TRUE(GNC_IS_MOCKTRANSACTION(trans));
-    ((MockTransaction*)trans)->set_description(desc);
+    gnc_mocktransaction(trans)->set_description(desc);
 }
 
 const char *
 xaccTransGetNotes (const Transaction *trans)
 {
-    EXPECT_TRUE(GNC_IS_MOCKTRANSACTION(trans));
-    if (GNC_IS_MOCKTRANSACTION(trans))
-        return ((MockTransaction*)trans)->get_notes();
-    else
-        return nullptr;
+    SCOPED_TRACE("");
+    const MockTransaction* mocktrans = gnc_mocktransaction(trans);
+    return mocktrans ? mocktrans->get_notes() : "";
 }
 
 void
 xaccTransSetNotes (Transaction *trans, const char *notes)
 {
     ASSERT_TRUE(GNC_IS_MOCKTRANSACTION(trans));
```

Comment:
No, this was intended. I left `ASSERT_TRUE` in all void functions, since it seems to simplify things. The alternative would be four code lines instead of two:
```cpp
SCOPED_TRACE("");
MockTransaction* mocktrans = gnc_mocktransaction(trans);
if (mocktrans)
    mocktrans->set_description(notes);
```

What do you think? If you like I can change it.

Reply:
Oops, there seems to be a copy&paste error in the code. Of course function `xaccTransSetNotes()` has to call mock function `mocktrans->set_notes()`. I'll correct this, too.

Reply:
> since it seems to simplify things.

OK.